WordPress/WordPressTest/MixpanelProxyTests.m

```diff
@@ -0,0 +1,193 @@
+#import <Specta/Specta.h>
+#define EXP_SHORTHAND
+#import <Expecta/Expecta.h>
+#import <OCMock/OCMock.h>
+#import <Mixpanel/Mixpanel.h>
+#import "MixpanelProxy.h"
+
+SpecBegin(MixpanelProxy)
+
+__block id mixpanelMock;
+__block id mixpanelPeopleMock;
+__block MixpanelProxy *mixpanelProxy;
+__block NSMutableDictionary *superProperties;
+
+beforeEach(^{
+    mixpanelMock = OCMClassMock([Mixpanel class]);
+    mixpanelPeopleMock = OCMClassMock([MixpanelPeople class]);
+    [OCMStub(ClassMethod([mixpanelMock sharedInstance])) andReturn:mixpanelMock];
+    [OCMStub([mixpanelMock people]) andReturn:mixpanelPeopleMock];
+    
+    superProperties = [NSMutableDictionary new];
+    [OCMStub([mixpanelMock currentSuperProperties]) andReturn:superProperties];
+    
+    mixpanelProxy = [MixpanelProxy new];
+});
+
+describe(@"registerInstanceWithToken:", ^{
+    
+    it(@"should register the token with Mixpanel", ^{
+        NSString *token = @"mixpanel-token";
+        OCMExpect(ClassMethod([mixpanelMock sharedInstanceWithToken:token]));
+        
+        [mixpanelProxy registerInstanceWithToken:token];
+        
+        OCMVerifyAll(mixpanelMock);
+    });
+});
+
+describe(@"currentSuperProperties", ^{
+    
+    it(@"returns the currentSuperProperties", ^{
+        superProperties[@"property"] = @YES;
+        
+        NSDictionary *currentSuperProperties = mixpanelProxy.currentSuperProperties;
+        
+        expect([currentSuperProperties[@"property"] boolValue]).to.beTruthy();
+    });
+});
+
+// Create helper to intercept super property dictionary
+__block NSDictionary *superPropertiesRegistering;
+void (^interceptSuperPropertyDictionary)() = ^{
+    [OCMStub([mixpanelMock registerSuperProperties:OCMOCK_ANY]) andDo:^(NSInvocation *invocation) {
+        __unsafe_unretained NSDictionary *props;
+        [invocation getArgument:&props atIndex:2];
+        
+        superPropertiesRegistering = props;
+    }];
+};
+
+describe(@"incrementSuperProperty:", ^{
+    
+    NSString *property = @"property";
+    it(@"should increment the super property", ^{
+        superProperties[property] = @10;
+        interceptSuperPropertyDictionary();
+        
+        [mixpanelProxy incrementSuperProperty:property];
+        
+        setAsyncSpecTimeout(1.0);
+        
+        // Sleeping briefly to give the MixpanelProxy time to kick off the async tasks that eventually register the super properties
+        [NSThread sleepForTimeInterval:0.1];
```

Comment:
Ugh I know this is necessary for the test but I really hate seeing sleep/waits in unit tests as it artificially slows the suite down. Is it possible to maybe use an XCTestExpectation or something of the sort to hook into the async wait? Ultimately 0.1 isn't too too long but this can add up over time.